osde2e/osd_metrics_exporter_tests.go

```diff
@@ -0,0 +1,125 @@
+// DO NOT REMOVE TAGS BELOW. IF ANY NEW TEST FILES ARE CREATED UNDER /osde2e, PLEASE ADD THESE TAGS TO THEM IN ORDER TO BE EXCLUDED FROM UNIT TESTS.
+//go:build osde2e
+// +build osde2e
+
+package osde2etests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	clustersmgmtv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/osde2e-common/pkg/clients/ocm"
+	"github.com/openshift/osde2e-common/pkg/clients/openshift"
+	"github.com/openshift/osde2e-common/pkg/clients/prometheus"
+	. "github.com/openshift/osde2e-common/pkg/gomega/assertions"
+	. "github.com/openshift/osde2e-common/pkg/gomega/matchers"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var _ = ginkgo.Describe("osd-metrics-exporter", ginkgo.Ordered, func() {
+	var (
+		clusterID         string
+		k8s               *openshift.Client
+		prom              *prometheus.Client
+		ocmClient         *ocm.Client
+		namespace         = "openshift-osd-metrics"
+		serviceName       = "osd-metrics-exporter"
+		deploymentName    = "osd-metrics-exporter"
+		rolePrefix        = "osd-metrics-exporter"
+		clusterRolePrefix = "osd-metrics-exporter"
+	)
+
+	ginkgo.BeforeAll(func(ctx context.Context) {
+		log.SetLogger(ginkgo.GinkgoLogr)
+
+		clusterID = os.Getenv("CLUSTER_ID")
+		Expect(clusterID).ShouldNot(BeEmpty(), "failed to find CLUSTER_ID environment variable")
+
+		var err error
+		k8s, err = openshift.New(ginkgo.GinkgoLogr)
+		Expect(err).ShouldNot(HaveOccurred(), "unable to setup k8s client")
+
+		prom, err = prometheus.New(ctx, k8s)
+		Expect(err).ShouldNot(HaveOccurred(), "unable to setup prometheus client")
+
+		ocmClient, err = ocm.New(ctx, os.Getenv("OCM_TOKEN"), ocm.Stage)
```

Comment:
Just a thought: we might want to eventually get the ocm environment from environment variable when we want to run the test suite against different ocm environments when working within progressive delivery. Maybe we can add a todo?